app/git_providers/github.rb

```diff
@@ -0,0 +1,69 @@
+require "http_link_header"
+require "json"
+require "net/http"
+require "uri"
+
+module Tint
+	module GitProviders
+		class Github
+			def initialize(token)
+				@token = token
+			end
+
+			def repositories
+				next_page = "/user/repos"
+				results = []
+
+				while next_page
+					next_page, page = repositories_page(next_page)
+					results += page
+				end
+
+				results
+			end
+
+		protected
+
+			def repositories_page(uri)
+				resp = request("GET", uri, per_page: 100)
+				raise resp unless resp.is_a?(Net::HTTPOK)
```

Comment:
When testing this I got a response of class `Net::HTTPUnauthorized` but it did not raise the exception as expected. It is a TypeError with the message "exception class/object expected". We could get around this with a custom exception class that takes in the response.